src/app/config.rs

```diff
@@ -409,6 +412,9 @@ impl<'a> TryFrom<&'a Strings> for Config {
             tls_controller: controller_namespace?,
         };
 
+        let proxy_id_template = strings.get(ENV_PROXY_ID)?.unwrap_or(String::new());
+        let proxy_id = proxy_id_template.replace(VAR_POD_NAMESPACE, &namespaces.pod);
```

Comment:
wait... what is this doing? why are we doing any special templating in the proxy?

Reply:
Specifically, all of this templating can be done via the kubernetes manifest:

```yaml
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  labels:
    app: env
  name: env
spec:
  selector:
    matchLabels:
      app: env
  template:
    metadata:
      labels:
        app: env
    spec:
      containers:
      - name: env
        command: ["sh", "-c"]
        args:
        - while true; do echo "id=${ID}" ; sleep 10 ; done
        image: debian:jessie
        env:
        - name: NS
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: ID
          value: "$(NAME).$(NS).suffix"
```